homeassistant/components/onvif/device.py

```diff
@@ -286,7 +286,20 @@ async def async_check_date_and_time(self) -> None:
     async def async_get_device_info(self) -> DeviceInfo:
         """Obtain information about this device."""
         device_mgmt = self.device.create_devicemgmt_service()
-        device_info = await device_mgmt.GetDeviceInformation()
+        manufacturer = None
+        model = None
+        firmware_version = None
+        serial_number = None
+        try:
+            device_info = await device_mgmt.GetDeviceInformation()
+        except (XMLParseError, XMLSyntaxError, TransportError) as ex:
+            # Some cameras have invalid UTF-8 in their device information
+            LOGGER.warning("%s: Failed to fetch device information: %s", self.name, ex)
```

Comment:
Do you need to handle all three errors?
From the corresponding issue it looks like only `TransportError` is necessary.

Reply:
Yes.  There is another camera that has the other ones